ltm/agent.py

```diff
@@ -76,43 +93,50 @@ def reply(self, user_message: str, agent_response: Optional[str] = None, cost_ca
 These tasks contains two things: a trigger condition, and a payload.
 The trigger conditions have been met for all task. So they can be safely disregarded. Now all you need to do is perform the tasks described in the "payload" fields. 
 
-You should append the result of your task to your reply to the query above
+You should append the result of your task to your reply to the query above.
 
-Append the results of your tasks to your current response.
-"""
+Append the results of your tasks to your current response."""
 
-        colour_print("CYAN", f"DEALING WITH USER MESSAGE: {user_message}")
         self.now = datetime.datetime.now()
 
-        keywords = self.keywords_for_message(user_message, cost_cb=cost_callback)
-
-        if not agent_response:
-            context = self.create_context(user_message, max_prompt_size=self.max_prompt_size, previous_interactions=0, cost_cb=cost_callback)
+        keywords = self.keywords_for_message(user_message)
 
+        if agent_response is None:
+            memories = self.collect_memories(user_message, previous_interactions=0)
         else:
-            context = [make_user_message(user_message)]
+            memories = []
 
         # Update task memory
-        self.add_task(deepcopy(context), cost_callback, user_message)
+        context = [make_system_message(self.system_message)] + memories + [
+            make_user_message(stamp_content(user_message, self.now, dt=self.now))
+        ]
+        self.add_task(deepcopy(context), user_message)
 
         # Get any tasks that might be due now.
         extra_tasks = self.progress_and_get_extra_tasks(context)
-        if len(extra_tasks) > 0:
-            context[-1]["content"] += perform_task.format(task=json.dumps(extra_tasks, indent=2))
+        if len(extra_tasks) > 0 or agent_response is None:
```

Comment:
This breaks prospective memory tests as it will only perform a task on messages where there are no trivia. The agent has to be able to perform tasks on any message.

Reply:
It will perform the task in both cases, but maybe this is a sign that the way it's put is maybe unintuitive. I can re-write it in a more sensible way.

Reply:
Oh yeah, its an `or` there!